src/db/sysdb_sudo.c

```diff
@@ -76,9 +76,18 @@ static errno_t sysdb_sudo_convert_time(const char *str, time_t *unix_time)
     for (format = formats; *format != NULL; format++) {
         /* strptime() may leave some fields uninitialized */
         memset(&tm, 0, sizeof(struct tm));
+        /* Let underlying implementation figure out DST */
+        tm.tm_isdst = -1;
         tret = strptime(str, *format, &tm);
         if (tret != NULL && *tret == '\0') {
-            *unix_time = mktime(&tm);
+            /* Convert broken-down time to local time */
+            if (tm.tm_gmtoff == 0) {
+                *unix_time = timegm(&tm);
+            } else {
+                long offset = tm.tm_gmtoff;
+                tm.tm_gmtoff = 0;
+                *unix_time = timegm(&tm) - offset;
+            }
```

Comment:
I wonder, can we simplify it and remove the first branch?

Reply:
I.e. just use

```c
                long offset = tm.tm_gmtoff;
                tm.tm_gmtoff = 0;
                *unix_time = timegm(&tm) - offset;
```

Reply:
@pbrezina sure you can but i just fail to see what would that accomplish exactly. the original is arguably easier to read and more efficient (i wont be surprised if the compiler opt generated the same for both in the end). either way the PR is open for edits so feel free to slice and dice it anyway you see fit.

Reply:
It's author's preference, I'm fine with it.